File: Proxy/Ruby/ProxyNetwork.rb
require_relative "INetwork"

class ProxyNetwork < INetwork
    def initialize(network)
        @network = network
        @bannedSites = [
            "abc.com",
            "xyz.com"
        ]
    end

    def connect(site)
        if @bannedSites.include?(site)
            puts "Access denied to #{site}"
        else
            @network.connect(site)
        end 
    end
end

File: Proxy/Javascript/ProxyNetwork.js
import INetwork from "./INetwork.js";

class ProxyNetwork extends INetwork {
  constructor(network) {
    super();
    this.network = network;
    this.bannedSites = [
      "abc.com",
      "xyz.com"
    ];
  }

  connect(site) {
    if (this.bannedSites.indexOf(site) === -1) {
      this.network.connect(site);
    } else {
      console.log(`Connection to ${site} is not allowed.`);
    }
  }
}

export default ProxyNetwork;